src/openeo_test_suite/tests/processes/metadata/test_process_metadata.py

```diff
@@ -0,0 +1,142 @@
+from typing import List
+import pytest
+from openeo_test_suite.lib.process_registry import ProcessData
+from openeo_test_suite.lib.process_selection import get_selected_processes
+import logging
+
+
+@pytest.fixture(scope="module")
+def api_processes(connection) -> List[dict]:
+    return connection.list_processes()
+
+
+def _get_test_id(val):
+    if isinstance(val, ProcessData):
+        return val.process_id
+
+
+@pytest.mark.parametrize(
+    "expected_process",
+    [p for p in get_selected_processes()],
+    ids=_get_test_id,
+)
+def test_process_metadata_functional(api_processes, expected_process, skipper):
+    """
+    Tests if the metadata of processes are correct, first tests if the process exists,
+    then tests if the parameters of processes are correct and finally tests if the return type of processes is correct.
+
+    Any process that has no metadata is skipped.
+
+    These are the functional parts of the process metadata e.g.: the parameters and return type.
+    """
+
+    skipper.skip_if_unsupported_process(expected_process.process_id)
+
+    actual_process = [
+        process
+        for process in api_processes
+        if process["id"] == expected_process.process_id
+    ][0]
+    # Tests if the parameters of processes are correct
+
+    expected_parameters = expected_process.spec.get("parameters", [])
+    actual_parameters = actual_process["parameters"]
+
+    if len(expected_parameters) > len(actual_parameters):
+        logging.warn(
+            f"Process {expected_process.process_id} has {len(expected_parameters)} expected parameters, but only {len(actual_parameters)} actual parameters"
+        )
+        for expected_parameter in expected_parameters[len(actual_parameters) :]:
+            assert expected_parameter.get("default", None) != None
```

Comment:
This is a bit confusing but I think this assert should check that the expected parameter has a null default.
Also note that with the current for loop'ed assert the error message will look quite confusing and non-actionable (because it says something about an expected parameter) to the test suite user.

Reply:
So as per the comment from Matthias https://github.com/Open-EO/openeo-test-suite/pull/44#discussion_r1471001093 
I am checking if the added or removed parameters have a default value, I was not checking for its value, but I've just changed it to checking for the key instead of checking if the value is not None (as null might be serialized to None)

I've also added explicit messages for both of these cases so the user can get actionable feedback.

Reply:
ok makes more sense now, thanks